pkg/policy/mapstate.go

```diff
@@ -400,17 +400,24 @@ type mapStateEntry struct {
 }
 
 // newMapStateEntry creates a map state entry.
-func newMapStateEntry(derivedFrom ruleOrigin, proxyPort uint16, priority ListenerPriority, deny bool, authReq AuthRequirement) mapStateEntry {
+func newMapStateEntry(
```

Comment:
I wonder if we can get rid of this constructor entirely? Changing all the call sites when a new optional parameter is added is frustrating.

Reply:
Agree, done in an additional commit.